qiskit/dagcircuit/dagcircuit.py

```diff
@@ -838,17 +838,18 @@ def compose(self, other, qubits=None, clbits=None, front=False, inplace=True):
     def reverse_ops(self):
         """Reverse the operations in the ``self`` circuit.
 
+        .. note::
+
+            The output reversed :class:`~.DAGCircuit` object will have shared
+            references for all operations and qubits to this dag.
```

Comment:
Qubits are immutable, so it's probably not worth mentioning them.  If we're worried about ownership here, it could be worth adding a `copy_operations` flag like we've got for `circuit_to_dag` and `dag_to_circuit`, and replace the `copy.copy` with a `copy.deepcopy` based on that.  I'm not certain if it's worth it, though.

Reply:
My first attempt at this locally was using a deepcopy instead. The deepcopy on its own ended up being slower than the previous implementation of the method. I figured this was an ok tradeoff because the ownership here wasn't clearly established before. But I don't feel super strongly about it either.